src/applications/personalization/profile360/util/index.js

```diff
@@ -8,3 +23,98 @@ export async function getData(apiRoute, options) {
     return { error };
   }
 }
+
+function hasErrorMessageKey(errors, errorKey) {
+  return errors.some(err =>
+    err.meta.messages.some(message => message.key === errorKey),
+  );
+}
+
+function hasErrorMessageText(errors, errorText) {
+  return errors.some(err =>
+    err.meta.messages.some(message =>
+      message.text.toLowerCase().includes(errorText.toLowerCase()),
+    ),
+  );
+}
+
+export function hasFlaggedForFraudError(errors) {
+  return (
+    hasErrorMessageKey(errors, ACCOUNT_FLAGGED_FOR_FRAUD_KEY) ||
+    hasErrorMessageKey(errors, PAYMENT_RESTRICTIONS_PRESENT_KEY) ||
+    hasErrorMessageKey(errors, ROUTING_NUMBER_FLAGGED_FOR_FRAUD_KEY)
+  );
+}
+
+export function hasInvalidRoutingNumberError(errors) {
+  let result = false;
+  if (hasErrorMessageKey(errors, INVALID_ROUTING_NUMBER_KEY)) {
+    result = true;
+  }
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    hasErrorMessageText(errors, 'Invalid Routing Number')
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+export function hasInvalidAddressError(errors) {
+  let result = false;
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    hasErrorMessageText(errors, 'address update')
+  ) {
+    result = true;
+  }
+  return result;
+}
+
+export function hasInvalidHomePhoneNumberError(errors) {
+  let result = false;
+  if (
+    hasErrorMessageKey(errors, GENERIC_ERROR_KEY) &&
+    (hasErrorMessageText(errors, 'night phone number') ||
+      hasErrorMessageText(errors, 'night area number'))
+  ) {
+    result = true;
+  }
+  return result;
+}
```

Comment:
Since these's kind of a pattern here, you could create a helper function, then have all of these leverage that instead -

`const hasInvalidHomePhoneNumberError = defineSpecialError([GENERIC_ERROR_KEY], ['night phone number', 'night area number'];`

Reply:
again, good call. I thought about refactoring things along those lines when I originally wrote these but then I forgot about it.

Reply:
I'm gonna make this change in a follow up.